src/IO/ADIOS/ADIOS2PreloadAttributes.cpp

```diff
@@ -102,7 +102,10 @@ namespace detail
                 {
                     numItems *= extent;
                 }
-                new( dest ) T[ numItems ]{};
+                for( size_t i = 0; i < numItems; ++i )
```

Comment:
That construct looks adventurous/expensive, why do we need this?

Reply:
Ah yes, should have written something on that one here too.
In most cases, `numItems` is 1, for our unit dimension things it is 7. I think, the largest instances are for the `VEC_STRING` datatype since those are represented as 2D char vectors, so you'll get some 2-digit number there.

> so it's probably not more expensive to loop here

I think it shouldn't be since this doesn't allocate anything.

Theoretically, we could also memset the whole thing to zero and use placement new only for non-basic datatypes, but that makes things very finicky to deal with and I don't think it's worth.

Also, I don't use `delete[]`, see `struct AttributeLocationDestroy` in the same file for the destructors.

But yeah, this file was tricky to get right